src/Components/Web.JS/src/Virtualize.ts

```diff
@@ -0,0 +1,77 @@
+export const Virtualize = {
+  init,
+  dispose,
+};
+
+const observersByDotNetId = {};
+
+function findClosestScrollContainer(element: Element | null): Element | null {
+  if (!element) {
+    return null;
+  }
+
+  const style = getComputedStyle(element);
+
+  if (style.overflowY !== 'visible') {
```

Comment:
Only supporting vertical scrolling is fine for a first cut of this feature. Longer term it's probably quite simple to extend it to support horizontal scrolling too. I think we just need to check whether the element's flex-direction is horizontal, and if so, look for `overflowX` instead.